receiver/kafkareceiver/header_extraction_test.go

```diff
@@ -0,0 +1,211 @@
+		nextConsumer: nextConsumer,
+		obsrecv:      obsrecv,
+	}
+	headers := []string{"headerKey1", "headerKey2"}
+	c.headerExtractor = &headerExtractor{
+		logger:  zap.NewNop(),
+		headers: headers,
+	}
+	groupClaim := &testConsumerGroupClaim{
+		messageChan: make(chan *sarama.ConsumerMessage),
+	}
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer close(groupClaim.messageChan)
+	testSession := testConsumerGroupSession{ctx: ctx}
+	require.NoError(t, c.Setup(testSession))
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		err = c.ConsumeClaim(testSession, groupClaim)
+		for _, logs := range nextConsumer.AllLogs() {
+			for i := 0; i < logs.ResourceLogs().Len(); i++ {
+				rs := logs.ResourceLogs().At(i)
+				val, ok := rs.Resource().Attributes().Get("kafka.header.headerKey1")
+				assert.Equal(t, ok, true)
+				assert.Equal(t, val.Str(), "headerValue1")
+				val, ok = rs.Resource().Attributes().Get("kafka.header.headerKey2")
+				assert.Equal(t, ok, true)
+				assert.Equal(t, val.Str(), "headerValue2")
+			}
+		}
+		assert.NoError(t, err)
+		wg.Done()
+	}()
+	groupClaim.messageChan <- &sarama.ConsumerMessage{
+		Headers: []*sarama.RecordHeader{
+			{
+				Key:   []byte("headerKey1"),
+				Value: []byte("headerValue1"),
+			},
+			{
+				Key:   []byte("headerKey2"),
+				Value: []byte("headerValue2"),
+			},
+		},
+		Value: []byte("Message"),
+	}
+	cancelFunc()
+	wg.Wait()
+
+}
+
+func TestHeaderExtractionMetrics(t *testing.T) {
+	obsrecv, err := obsreport.NewReceiver(obsreport.ReceiverSettings{ReceiverCreateSettings: receivertest.NewNopCreateSettings()})
+	nextConsumer := new(consumertest.MetricsSink)
+	c := metricsConsumerGroupHandler{
+		unmarshaler:  newPdataMetricsUnmarshaler(&pmetric.ProtoUnmarshaler{}, defaultEncoding),
+		logger:       zap.NewNop(),
+		ready:        make(chan bool),
+		nextConsumer: nextConsumer,
+		obsrecv:      obsrecv,
+	}
+	headers := []string{"headerKey1", "headerKey2"}
+	c.headerExtractor = &headerExtractor{
+		logger:  zap.NewNop(),
+		headers: headers,
+	}
+	groupClaim := &testConsumerGroupClaim{
+		messageChan: make(chan *sarama.ConsumerMessage),
+	}
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer close(groupClaim.messageChan)
+	testSession := testConsumerGroupSession{ctx: ctx}
+	require.NoError(t, c.Setup(testSession))
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		err = c.ConsumeClaim(testSession, groupClaim)
+		for _, metric := range nextConsumer.AllMetrics() {
+			for i := 0; i < metric.ResourceMetrics().Len(); i++ {
+				rs := metric.ResourceMetrics().At(i)
+				val, ok := rs.Resource().Attributes().Get("kafka.header.headerKey1")
+				assert.Equal(t, ok, true)
+				assert.Equal(t, val.Str(), "headerValue1")
+				val, ok = rs.Resource().Attributes().Get("kafka.header.headerKey2")
+				assert.Equal(t, ok, true)
+				assert.Equal(t, val.Str(), "headerValue2")
+			}
+		}
+		assert.NoError(t, err)
+		wg.Done()
+	}()
+	ld := testdata.GenerateMetricsOneMetric()
```

Comment:
It looks like these are repeating, could you extract these out to a setup function to reduce the complexity of setting up the tests?